packages/x-charts/src/BarChart/checkScaleErrors.ts

```diff
@@ -0,0 +1,54 @@
+import { DEFAULT_X_AXIS_KEY, DEFAULT_Y_AXIS_KEY } from '../constants';
+import { AxisDefaultized, isBandScaleConfig, isPointScaleConfig } from '../models/axis';
+import { SeriesId } from '../models/seriesType/common';
+
+const getXOrY = (isVertical: boolean, isContinuous: boolean, x: string, y: string) => {
+  if (isVertical) {
```

Comment:
It took me some time to understand `isVertical` was not about the axis, but the bar layout

```suggestion
const getXOrY = (verticalLayout: boolean, isContinuous: boolean, x: string, y: string) => {
  if (verticalLayout) {
```

Reply:
Just found another solution by passing `axisDirection: 'x' | 'y'` to the `getAxisMessage` Not sure it's a good idea. Just sharing